flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/utils/TypeTransformer.java

```diff
@@ -0,0 +1,542 @@
+import io.ballerina.compiler.api.symbols.FunctionTypeSymbol;
+import io.ballerina.compiler.api.symbols.FutureTypeSymbol;
+import io.ballerina.compiler.api.symbols.IntersectionTypeSymbol;
+import io.ballerina.compiler.api.symbols.MapTypeSymbol;
+import io.ballerina.compiler.api.symbols.ObjectTypeSymbol;
+import io.ballerina.compiler.api.symbols.Qualifier;
+import io.ballerina.compiler.api.symbols.RecordTypeSymbol;
+import io.ballerina.compiler.api.symbols.ResourceMethodSymbol;
+import io.ballerina.compiler.api.symbols.StreamTypeSymbol;
+import io.ballerina.compiler.api.symbols.SymbolKind;
+import io.ballerina.compiler.api.symbols.TypeDefinitionSymbol;
+import io.ballerina.compiler.api.symbols.TypeDescTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.api.symbols.UnionTypeSymbol;
+import io.ballerina.compiler.syntax.tree.RecordFieldWithDefaultValueNode;
+import io.ballerina.compiler.syntax.tree.RecordTypeDescriptorNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.flowmodelgenerator.core.model.Function;
+import io.ballerina.flowmodelgenerator.core.model.Member;
+import io.ballerina.flowmodelgenerator.core.model.ModuleInfo;
+import io.ballerina.flowmodelgenerator.core.model.NodeKind;
+import io.ballerina.flowmodelgenerator.core.model.TypeData;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.Module;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class TypeTransformer {
+    private final Module module;
+    private final ModuleInfo moduleInfo;
+    private Map<String, RecordTypeDescriptorNode> recordTypeDescNodes;
+
+    public TypeTransformer(Module module) {
+        this.module = module;
+        this.moduleInfo = ModuleInfo.from(module.descriptor());
+    }
+
+    public Object transform(ClassSymbol classSymbol) {
+        TypeData.TypeDataBuilder typeDataBuilder = new TypeData.TypeDataBuilder();
+        typeDataBuilder
+                .codedata()
+                    .node(NodeKind.CLASS)
+                    .stepOut()
+                .properties()
+                    .qualifiers(classSymbol.qualifiers().stream().map(Qualifier::getValue).toList(), true, true, true)
+                    .isArray("false", true, true, true)
+                    .arraySize("", false, false, false)
+                    .stepOut();
+
+        // inclusions
+        List<String> includes = new ArrayList<>();
+        classSymbol.typeInclusions().forEach(typeInclusion -> {
+            includes.add(CommonUtils.getTypeSignature(typeInclusion, this.moduleInfo));
+        });
+        typeDataBuilder.includes(includes);
+
+        Member.MemberBuilder memberBuilder = new Member.MemberBuilder();
+
+        // class fields
+        Map<String, Member> fieldMembers = new HashMap<>();
```

Comment:
We can have a one single function to handle fields and function in both objects and classes.